### PR TITLE
[#SALTY] GRAB NERF [FUCK] [BALANCE] [#GRABDATNERF]

### PR DESCRIPTION
**Pull request by Iamgoofball**
# oh you fucking know i'm goddamn salty aaaaaaaaaaaaa

:cl: Iamsaltball
rscadd: GRAB SUPER NERFED
rscadd: IT'S WAY EASIER TO ESCAPE GRABS NOW
rscadd: IT TAKES WAY LONGER TO UPGRADE GRABS NOW
rscadd: YOU CAN ACTUALLY FUCKING ESCAPE GRABS NOW TOO #WOW #WHOA 
rscadd: GRAB TEXT IS ALL BIG BOLD AND RED MUCH LIKE YOUR MOM LAST NIGHT
experimental: GIT GUDDERS GO FUCK YOURSELVES YOU CAN'T GET GOOD AGAINST "LOL INSTANT PERMASTUN THAT ISN'T OBVIOUS IN CHAT LIKE THE REST OF THE STUNS"
/:cl:

### CODE REVIEW

**Comment by Cheridan** [comment]
FYI unescapable grabs are a bug.

**Comment by Iamgoofball** [comment]
yes, and even if you fix the bug, grab escape changes are SO RETARDEDLY LOW

**Comment by PKPenguin321** [comment]
Iamgoofball more like "iamgayball"

**Comment by Fox-McCloud** [comment]
Grabs are ridiculous, yes :+1: 

This may be a bit much, but I definitely support them getting nerfed.

**Comment by KazeEspada** [comment]
:+1: agree

**Comment by Zilenan91** [comment]
Yep :+1: 

It's more efficient to grab someone than it is disarming and actually hitting someone with a weapon

**Comment by PKPenguin321** [comment]
At least leave the upgrade times alone. It is already practically impossible to choke somebody out.

**Comment by AlexanderUlanH** [comment]
:-1: once again Iamgoofball tries his hand at the 'nerf it so hard it might just as well not exist' approach.
Make grab resisting a do_after and make it prevent speech and you've got something.
In fact, why not just make the grab upgrades do_afters as well?
Choking is a fun feature and serves the important role of giving people the ability to cleanly murder people they can restrain. People should know when they're being grappled with, and should have a fair chance to break out of the grapple. But it should be deadly once they're unconscious and the grappler should have a fair chance to interrupt escape attempts.

**Comment by AdamElTablawy** [comment]
I'd agree with all of these except the upgrade timers and making chokeholds 0.5 rather than 1 oxy.

It's already impossible to choke someone out without stunning them somehow unless they're completely braindead, and if they're stunned they're already toast.

I'd leave upgrade timers alone and oxydamage alone, everything else is fine.

**Comment by Iamgoofball** [comment]
>It's already impossible to choke someone out without stunning
you realize chokeholds stun right

**Comment by Zilenan91** [comment]
Yeah and you can upgrade from a passive to aggressive grab near instantly. It's more effective to grab someone than toolbox them.

**Comment by AlexanderUlanH** [comment]
Grab upgrading has a cooldown after the first upgrade. If you have a table nearby, it's more effective to tablepush then toolbox, which is intended, but you can't get someone who's aware into a chokehold. The problem is that if you don't react in time you're dead.

**Comment by PKPenguin321** [comment]
Yeah, you can't just waltz up to somebody and chokehold them. A chokehold is like handcuffs, if you don't stun them first they can just walk away.

**Comment by Xhuis** [comment]
I support the escaping of grabs, but the rest of the changes are unnecessary. Unescapable permastun grabs were the main problem with them, and it already doesn't make sense at how slowly you can grab someone - with this change it'd be like time was being slowed to a crawl if you were grabbing someone because your hands were moving at the speed of molasses up a hill.

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -228,16 +228,16 @@
...
+					if(!prob(75))
 						return 1
-					mob.visible_message("<span class='warning'>[mob] has broken free of [G.assailant]'s grip!</span>")
+					mob.visible_message("<span class='danger'>[mob] has broken free of [G.assailant]'s grip!</span>")
 					qdel(G)
 
 				if(GRAB_NECK)
 					move_delay = world.time + 10
-					if(!prob(5))
+					if(!prob(50))
```
Comment (Xhuis) [inline code/modules/mob/mob_movement.dm]: _Slightly_ too high in my opinion; maybe 40? Either way it's much better than the ridiculously low chance it has now

Comment (Zilenan91) [inline code/modules/mob/mob_movement.dm]: Keep in mind they can continue choking you while moving, which interrupts your resisting.

Comment (AlexanderUlanH) [inline code/modules/mob/mob_movement.dm]: If we're going from "Inescapable permastun" to "Inescapable pemastun as long as the choker walks back and forth while choking" this is a step backwards

Comment (PKPenguin321) [inline code/modules/mob/mob_movement.dm]: Uhh you guys do know you can hold any movement key while being neckgrabbed and proc an escape attempt like 10,000 times in one second, right?

Comment (Zilenan91) [inline code/modules/mob/mob_movement.dm]: Don't escape attempts take time though? It doesn't matter if you can proc 10000 of them when none of htem will succeed

**Comment by Niknakflak** [comment]
I think just getting rid of the grab upgrade hotkey would fix this tbh

File: code/modules/mob/mob_movement.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/modules/mob/mob_movement.dm]: No, they are instant. Look at the code you are commenting on. Do you see a do_after?

**Comment by bgobandit** [comment]
rip mekhi

**Comment by TehZombehz** [comment]
Somebody had the guts to do it aaaaaaaaaaaaaaaaaa <3

**Comment by Bawhoppen** [comment]
This is completely ripping out and throwing away a key gameplay mechanic and will throw off the meta entirely.
-
-
-
-
-
-
-
-
-
-
:+1:

**Comment by PKPenguin321** [comment]
so can we talk about how you can literally break out of any level of grab except straight up choking instantly with this before we yolomerge it out of salt?

**Comment by lzimann** [comment]
have you tried... getting good?

**Comment by PKPenguin321** [comment]
stop memeing me

**Comment by Bawhoppen** [comment]
meme?

**Comment by PKPenguin321** [comment]
no but seriously the code for this is actually literally busted, you can break out of any grab instantly

**Comment by Bawhoppen** [comment]
@PKPenguin321 memememememememememememe

**Comment by Aloraydrel** [comment]
git good lmao. If you get yourself in a choke hold it's 90% your fault

**Comment by KazeEspada** [comment]
Nah, usually it isnt. Usually its someone using hotkeys and you arent.

**Comment by PKPenguin321** [comment]
>Nah, usually it isnt. Usually its someone using hotkeys and you arent.
 
nigga just walk away like just click on your arrow keys like damn haha

**Comment by lzimann** [comment]
if you can't walk away from a guy trying to neckgrab you, damn son, you deserve being grabbed

**Comment by Robustin** [comment]
What meme is this?

**Comment by lzimann** [comment]
@Robustin someone didn't get good and wants to nerf shit

**Comment by Zilenan91** [comment]
To be fair neck grabs are like timespace distortions sometimes. Bryce had me in an aggressive grab in the locker rooms near the washing machine and I managed to run away to the door of the locker rooms before I was literally teleported into his grip because BYond

**Comment by Fox-McCloud** [comment]
>To be fair neck grabs are like timespace distortions sometimes. Bryce had me in an aggressive grab in the locker rooms near the washing machine and I managed to run away to the door of the locker rooms before I was literally teleported into his grip because BYond

Sounds like grabs aren't garbage collecting 100% of the time.

**Comment by Zilenan91** [comment]
Well I mean

I don't think you literally warping 5+ tiles away into a permastun is a good feature

**Comment by PKPenguin321** [comment]
>I don't think you literally warping 5+ tiles away into a permastun is a good feature

Well I mean
In the situation you described you can just hold an arrow key and break out near instantly so you can barely call it a stun at all

**Comment by Zilenan91** [comment]
Even after you break out you're still stunned more than long enough to cuff, or be beaten to death by really anything stronger than a wrench.

**Comment by PKPenguin321** [comment]
hahaha no, unless you get hyper omega super cucked by RNG. proccing 10000 attempts at getting away by holding an arrow key with a 5 percent chance each time is a much better chance to escape than you'd imagine.

**Comment by Iamgoofball** [comment]
You realize your move delay gets cranked up super high, right?

**Comment by lzimann** [comment]
WHAT THE FUCK WHY STOP MERGING STUPID PRS

**Comment by Iamgoofball** [comment]
# S
# A
# L
# T

**Comment by Lumipharon** [comment]
why the actual fuck did this get merged?

Only choke holds can't be broken, if you let them do that, that involves staying still for like 10 seconds while they stroke their cock and slowly tighten their grip.

**Comment by Archie700** [comment]
>this was actually merged

did you have a brainfart

**Comment by Eaglendia** [comment]
What a joke

**Comment by optimumtact** [comment]
feel free to make your own server if you believe you can do better

:^)

**Comment by PKPenguin321** [comment]
what the fuck cheridan why

**Comment by Dawgas** [comment]
too bad that the mythical good player that people want to play with is more likely to leave ss13 entirely because of changes like this made to codebases than take time out of their busy lives to learn how to navigate the git and learn to code for byond

this is the exact kind of reason that we can't have nice things
brb starting a headcoder campaign with my qualification of not being terminally retarded and understanding the value of having a stable codebase

**Comment by Iamgoofball** [comment]
@Dawgas why dont you make a PR to rebalance it yourself

you have a github account

you know how to code

be the change you want to be

**Comment by PKPenguin321** [comment]
>why dont you make a PR to rebalance it yourself

"le kneejerk revert XD"

**Comment by Iamgoofball** [comment]
a revert isnt a rebalance

a rebalance would be adjusting numbers

**Comment by PKPenguin321** [comment]
tell that to the guy that'll close the PR :confused:

**Comment by Iamgoofball** [comment]
no, because a proper rebalance isn't a kneejerk revert

this got closed due to being a kneejerk revert

i can bet you 5 dollars it wont get closed instantly, and if it does I'll reopen the PR for you

**Comment by Bawhoppen** [comment]
stop spamming my email holy shit

**Comment by Archie700** [comment]
Can we just have it tested for a few rounds before the revert comes in

**Comment by Iamgoofball** [comment]
1. this isnt a test

2. a few rounds doesn't gather data from all timezones of ss13 players and all weekly players

we get different players on different days

we use week long tests because that hits all timezones, and all players from monday to sunday

**Comment by PKPenguin321** [comment]
guys can i give humans +100 max health so that they have more of a distinction between them and other mutant races it'll just be a test c'mon guys

**Comment by Archie700** [comment]
Can we lock this because this is becoming a shitpost haven for everyone

**Pull request merged**